src/fpm/manifest/profiles.f90

```diff
@@ -0,0 +1,732 @@
+      subroutine validate_compiler_name(compiler_name, is_valid)
+        character(len=:), allocatable, intent(in) :: compiler_name
+        logical, intent(out) :: is_valid
+        select case(compiler_name)
+          case("gfortran", "ifort", "ifx", "pgfortran", "nvfrotran", "flang", "caf", &
+                        & "f95", "lfortran", "lfc", "nagfor", "crayftn", "xlf90", "ftn95")
+            is_valid = .true.
+          case default
+            is_valid = .false.
+        end select
+      end subroutine validate_compiler_name
+        
+      subroutine validate_os_name(os_name, is_valid)
+        character(len=:), allocatable, intent(in) :: os_name
+        logical, intent(out) :: is_valid
+        select case (os_name)
+          case ("linux", "macos", "windows", "cygwin", "solaris", "freebsd", &
+                          & "openbsd", "unknown", "UNKNOWN")
+            is_valid = .true.
+          case default
+            is_valid = .false.
+        end select
+      end subroutine validate_os_name
+
+      subroutine match_os_type(os_name, os_type)
+        character(len=:), allocatable, intent(in) :: os_name
+        integer, intent(out) :: os_type
+        select case (os_name)
+          case ("linux");   os_type = OS_LINUX
+          case ("macos");   os_type = OS_WINDOWS
+          case ("cygwin");  os_type = OS_CYGWIN
+          case ("solaris"); os_type = OS_SOLARIS
+          case ("freebsd"); os_type = OS_FREEBSD
+          case ("openbsd"); os_type = OS_OPENBSD
+          case ("all");     os_type = OS_ALL
+          case default;     os_type = OS_UNKNOWN
+        end select
+      end subroutine match_os_type
+
+      subroutine get_flags(profile_name, compiler_name, os_type, key_list, table, profiles, profindex, error)
+
+        !> Name of profile
+        character(len=:), allocatable, intent(in) :: profile_name
+
+        !> Name of compiler
+        character(len=:), allocatable, intent(in) :: compiler_name
+
+        !> OS type
+        integer, intent(in) :: os_type
+
+        !> List of keys in the table
+        type(toml_key), allocatable, intent(in) :: key_list(:)
+
+        !> Table containing OS tables
+        type(toml_table), pointer, intent(in) :: table
+
+        !> List of profiles
+        type(profile_config_t), allocatable, intent(inout) :: profiles(:)
+
+        !> Index in the list of profiles
+        integer, intent(inout) :: profindex
+
+        !> Error handling
+        type(error_t), allocatable, intent(out) :: error
+
+        character(len=:), allocatable :: flags, c_flags, link_time_flags, key_name, file_name, file_flags
+        type(toml_table), pointer :: files
+        type(toml_key), allocatable :: file_list(:)
+        type(file_scope_flag), allocatable :: file_scope_flags(:)
+        integer :: ikey, ifile, stat
+        logical :: is_valid
+
+        if (size(key_list).ge.1) then
+          do ikey=1,size(key_list)
+            key_name = key_list(ikey)%key
```

Comment:
Why do we have to iterate over a list of keys here? All required key names in a profile are known at compile time.

Reply:
This is supposed to be ready for further error handling. We had a discussion about whether an error should occur if the table contains other key names.

Reply:
Also, I think that this is the easiest way how to deal with the error handling of the fields as they are. They do not need to be defined, but if they are, they need to be just key-value pairs.

Reply:
This still needs to be addressed.

Reply:
I still think that this is the easiest solution, if you want to suggest anything else, please do.